src/SilverStripe/LocalGit/GitRepository.php

```diff
@@ -14,12 +14,18 @@ class GitRepository extends ReadonlyGitRepository {
 	protected $cleanup = true;
 
 	/**
+	 * @var string
+	 */
+	private $knownHostsFile;
+
+    /**
 	 * @param string $gitUrl
 	 * @param string $revision
 	 * @param string|null $identityFile
 	 * @param string|null $localPath Specify local path to clone repo. Defaults to auto-generated path in temp
+	 * @param string|null $knownHostsFile Path to custom known_hosts. Must be read/writable.
 	 */
-	public function __construct($gitUrl, $revision = 'master', $identityFile = null, $localPath = null) {
+	public function __construct($gitUrl, $revision = 'master', $identityFile = null, $localPath = null, $knownHostsFile = null) {
```

Comment:
This signature is a bit unwieldy now, can we set the optional argument via method setters instead?

Reply:
I can't without heavy refactoring (not a good reason, I know, but...). __construct calls cloneTempRepo, and I have to have knownHostsFile by then.

Reply:
right, I think I've learned a lesson to never to create a class to do 'work' in a constructor. :/